### PR TITLE
Merchant Payments

### PR DESCRIPTION
**Pull request by dckesler**
### Description

In preparation for celo-connect where we would like the conference participants to pay with valora using the merchant-payments API, here is the PR adding that behaviour (under a feature flag).

### Other changes

Nothing normally

### Tested

https://user-images.githubusercontent.com/8698308/157790697-c4d318a1-d2a1-4f2e-9e03-08e0f9d4879b.mov

### How others should test

We have a test shop here: https://sandbox-celo-demo-store.demo.firstdag.com/

### Backwards compatibility

This new feature is behind a feature flag, should be 100% backwards compatible.

### CODE REVIEW

**Review by gnardini — APPROVED** [review]
Approving with the same comments as https://github.com/valora-inc/wallet/pull/1976#pullrequestreview-904737906

File: packages/mobile/test/schemas.ts [hunk 1]
```diff
@@ -1035,6 +1036,18 @@ export const v38Schema = {
   },
 }
 
+export const v39Schema = {
+  ...v38Schema,
+  _persist: {
+    ...v38Schema._persist,
+    version: 38,
```
Comment (jeanregisser) [inline packages/mobile/test/schemas.ts:1043]: This must be changed to 39.
There's a test around this, but you found a loophole 😅

**Review by jeanregisser — CHANGES_REQUESTED** [review]

File: packages/mobile/test/schemas.ts (hunk 1 above)
Comment (nicolasbrugneaux) [reply packages/mobile/test/schemas.ts]: done! :) good catch

**Review by jeanregisser — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT, For android platform Latest Android Alfajores build V1.28.1 is not displayed in play store , So we have blocked the test cases for android platform.

**Test Rail Links:**
**Android 10:** https://valoraapp.testrail.io/index.php?/tests/view/16582

**Devices:**  Samsung Galaxy M40(10.0)

**CC:** @dckesler 

Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified this task on Latest Test flight alfa jores build V1.28.1 & update the status in test rails. 

**Note :** User is able to complete the whole flow on sending amount by deeplinking, but error appears when user tries to scan qr from the merchant website url, we have raised a bug for it.



**New Bug:** [Link](https://github.com/valora-inc/wallet/issues/2156)

**Test Rail Links:**

**iOS 14:** https://valoraapp.testrail.io/index.php?/tests/view/17649&group_by=cases:section_id&group_order=asc&group_id=40

**Devices:** iPhone 12(14.7.1)

**CC:** @dckesler
Thanks..!!

**Comment by ValoraQA** [comment]
Hey @MuckT can you provide an update for the observation mentioned in the above comment

**Pull request merged**